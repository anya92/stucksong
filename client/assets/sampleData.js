export const recentlyPlayedTracks = [
  {
    album: 'Americana',
    artist: 'The Offspring',
    id: '4EchqUKQ3qAQuRNKmeIpnf',
    image: 'https://i.scdn.co/image/c77a1f3a2de72c2a8fde481a016376ca8ef81b33',
    played_at: '2018-08-14T18:50:09.757Z',
    title: 'The Kids Aren\'t Alright',
    uri: 'spotify:track:4EchqUKQ3qAQuRNKmeIpnf',
  },
  {
    album: 'No Pads, No Helmets...Just Balls (15th Anniversary Tour Edition)',
    artist: 'Simple Plan',
    id: '6otiaV2fagE3s8IvP6WkwG',
    image: 'https://i.scdn.co/image/9f6a521519080242f2de1c2b8873eec5372f7a6a',
    played_at: '2018-08-14T18:46:04.608Z',
    title: 'I\'m Just A Kid',
    uri: 'spotify:track:6otiaV2fagE3s8IvP6WkwG',
  },
  {
    album: 'All Killer, No Filler',
    artist: 'Sum 41',
    id: '1HNE2PX70ztbEl6MLxrpNL',
    image: 'https://i.scdn.co/image/4fc4f0bb924338f85e0fe7c899cde32ec5e9f10d',
    played_at: '2018-08-14T18:42:32.417Z',
    title: 'In Too Deep',
    uri: 'spotify:track:1HNE2PX70ztbEl6MLxrpNL',
  },
];

export const topArtists = [
  {
    genres: [
      'dance pop',
      'folk-pop',
      'indie pop',
      'indie r&b',
      'indietronica',
      'metropopolis',
      'pop',
    ],
    id: '4Ui2kfOqGujY81UcPrb5KE',
    image: 'https://i.scdn.co/image/a20acc6267aec9d08aa32d05ae2fb8647943f81a',
    name: 'HAIM',
    uri: 'spotify:artist:4Ui2kfOqGujY81UcPrb5KE',
  },
  {
    genres: [
      'chamber pop',
      'indie folk',
      'shimmer pop',
    ],
    id: '46CitWgnWrvF9t70C2p1Me',
    image: 'https://i.scdn.co/image/8b6e70aafad73059422920cbe528ecbcdff74018',
    name: 'Daughter',
    uri: 'spotify:artist:46CitWgnWrvF9t70C2p1Me',
  },
  {
    genres: [
      'alt-indie rock',
      'garage rock',
      'indie rock',
      'indietronica',
      'modern alternative rock',
      'modern rock',
    ],
    id: '3btzEQD6sugImIHPMRgkwV',
    image: 'https://i.scdn.co/image/f54c5e6b3919940e204406ef6846ee8e197f8217',
    name: 'Wolf Alice',
    uri: 'spotify:artist:3btzEQD6sugImIHPMRgkwV',
  },
];

export const topTracks = [
  {
    album: 'Gone Now',
    artist: 'Bleachers',
    id: '3ySU5vwQB33iGulwcUL9qQ',
    image: 'https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb',
    title: 'Don\'t Take The Money',
    uri: 'spotify:track:3ySU5vwQB33iGulwcUL9qQ',
  },
  {
    album: 'Something To Tell You',
    artist: 'HAIM',
    id: '20QoFTI0fp2dNNexlHDhTe',
    image: 'https://i.scdn.co/image/9925a5f35b0a0e7b6f16e8f12c80cb5b4f015d57',
    title: 'Walking Away',
    uri: 'spotify:track:20QoFTI0fp2dNNexlHDhTe',
  },
  {
    album: 'Big Mess',
    artist: 'Grouplove',
    id: '1Tcd9flKIOxXWqR28aDP3U',
    image: 'https://i.scdn.co/image/781fa2da2c0dc8fc64f093dd753d4acfd6647db1',
    title: 'Don\'t Stop Making It Happen',
    uri: 'spotify:track:1Tcd9flKIOxXWqR28aDP3U',
  },
];

export const playlist = {
  name: 'Top Tracks',
  description: '',
  image: 'https://mosaic.scdn.co/640/129c6e4a622dc50c0802042780f92631a9dcc14056cc004f3ff47cff0ad8fcf05461730a193d8d8263b1ad734c51e06c7faee7fc95fd3abf0bbb9bbe6f4c08a3f6b73b062d6ea8384add3e3889814448',
  url: 'https://open.spotify.com/playlist/6yZesRjpu9OPKCM77N2EQt',
  tracks: topTracks,
};
