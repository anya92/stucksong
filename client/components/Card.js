import React from 'react';
import {
  string,
  number,
  object,
} from 'prop-types';

import {
  CardImage,
  CardImageLink,
  CardInfo,
  CardTitle,
  CardText,
  CardArtist,
  CardAlbum,
} from '../styles/cards';

const Card = ({ data, type, index }) => (
  <div>
    <CardImage>
      <CardImageLink>
        <a
          href={`https://open.spotify.com/${type}/${data.id}`}
          // href={`${data.uri}`}
          target="_blank"
          rel="noopener noreferrer"
        >play
        </a>
      </CardImageLink>
      <img src={data.image} alt={type === 'artists' ? data.name : data.album} />
    </CardImage>
    <CardInfo>
      <CardTitle>
        <span>{index + 1}.</span> {data.title}
      </CardTitle>
      <CardText>
        <CardArtist>{data.artist}</CardArtist>
        <CardAlbum>{data.album}</CardAlbum>
      </CardText>
    </CardInfo>
  </div>
);

Card.propTypes = {
  data: object.isRequired,
  type: string.isRequired,
  index: number.isRequired,
};

export default Card;
