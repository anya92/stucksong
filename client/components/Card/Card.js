import React from 'react';
import {
  string,
  number,
  shape,
  oneOf,
} from 'prop-types';
import format from 'date-fns/format';

import {
  CardImage,
  CardImageLink,
  CardImageDate,
  CardInfo,
  CardTitle,
  CardText,
  CardArtist,
  CardAlbum,
  CardGenres,
} from '../../styles/cards';

const formatGenres = (dataGenres) => {
  let genres = '';
  dataGenres.slice(0, 3).forEach((genre) => { genres += `${genre}, `; });
  return genres.slice(0, -2);
};

const Card = ({ data, type, index }) => (
  <div>
    <CardImage>
      <CardImageLink>
        <a
          href={`https://open.spotify.com/${type === 'artist' ? 'artist' : 'track'}/${data.id}`}
          // href={`${data.uri}`}
          target="_blank"
          rel="noopener noreferrer"
        > { type === 'artist' ? 'open' : 'play' }
        </a>
      </CardImageLink>
      <img src={data.image} alt={type === 'artist' ? data.name : data.album} />
      {
        type === 'recently-track'
        && <CardImageDate>{ format(data.played_at, 'DD MMM - HH:mm') }</CardImageDate>
      }
    </CardImage>
    <CardInfo>
      <CardTitle>
        { type !== 'recently-track' && <span>{index + 1}.</span> }
        { type === 'artist' ? data.name : data.title }
      </CardTitle>
      <CardText>
        {
          type === 'artist'
            ? <CardGenres>{ formatGenres(data.genres) }</CardGenres>
            : (
              <React.Fragment>
                <CardArtist>{data.artist}</CardArtist>
                <CardAlbum>{data.album}</CardAlbum>
              </React.Fragment>
            )
        }
      </CardText>
    </CardInfo>
  </div>
);

Card.propTypes = {
  data: shape({}).isRequired,
  type: oneOf(['track', 'artist', 'recently-track']).isRequired,
  index: number.isRequired,
};

export default Card;
