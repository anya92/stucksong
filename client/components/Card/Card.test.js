import React from 'react';
import { shallow } from 'enzyme';

import Card from './Card';
import * as data from '../../assets/sampleData';

describe('Card component', () => {
  it('renders Card with top track data correctly', () => {
    const mockProps = {
      data: data.topTracks[0],
      type: 'track',
      index: 0,
    };
    const wrapper = shallow(<Card {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Card with top artist data correctly', () => {
    const mockProps = {
      data: data.topArtists[0],
      type: 'artist',
      index: 0,
    };
    const wrapper = shallow(<Card {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Card with recently played track data correctly', () => {
    const mockProps = {
      data: data.recentlyPlayedTracks[0],
      type: 'recently-track',
      index: 0,
    };
    const wrapper = shallow(<Card {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
