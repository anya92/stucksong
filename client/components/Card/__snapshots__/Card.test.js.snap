// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card with recently played track data correctly 1`] = `
<div>
  <glamorous(div)>
    <glamorous(div)>
      <a
        href="https://open.spotify.com/track/4EchqUKQ3qAQuRNKmeIpnf"
        rel="noopener noreferrer"
        target="_blank"
      >
         
        play
      </a>
    </glamorous(div)>
    <img
      alt="Americana"
      src="https://i.scdn.co/image/c77a1f3a2de72c2a8fde481a016376ca8ef81b33"
    />
    <glamorous(div)>
      14 Aug - 20:50
    </glamorous(div)>
  </glamorous(div)>
  <glamorous(div)>
    <glamorous(div)>
      The Kids Aren't Alright
    </glamorous(div)>
    <glamorous(div)>
      <glamorous(div)>
        The Offspring
      </glamorous(div)>
      <glamorous(div)>
        Americana
      </glamorous(div)>
    </glamorous(div)>
  </glamorous(div)>
</div>
`;

exports[`Card component renders Card with top artist data correctly 1`] = `
<div>
  <glamorous(div)>
    <glamorous(div)>
      <a
        href="https://open.spotify.com/artist/4Ui2kfOqGujY81UcPrb5KE"
        rel="noopener noreferrer"
        target="_blank"
      >
         
        open
      </a>
    </glamorous(div)>
    <img
      alt="HAIM"
      src="https://i.scdn.co/image/a20acc6267aec9d08aa32d05ae2fb8647943f81a"
    />
  </glamorous(div)>
  <glamorous(div)>
    <glamorous(div)>
      <span>
        1
        .
      </span>
      HAIM
    </glamorous(div)>
    <glamorous(div)>
      <glamorous(div)>
        dance pop, folk-pop, indie pop
      </glamorous(div)>
    </glamorous(div)>
  </glamorous(div)>
</div>
`;

exports[`Card component renders Card with top track data correctly 1`] = `
<div>
  <glamorous(div)>
    <glamorous(div)>
      <a
        href="https://open.spotify.com/track/3ySU5vwQB33iGulwcUL9qQ"
        rel="noopener noreferrer"
        target="_blank"
      >
         
        play
      </a>
    </glamorous(div)>
    <img
      alt="Gone Now"
      src="https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb"
    />
  </glamorous(div)>
  <glamorous(div)>
    <glamorous(div)>
      <span>
        1
        .
      </span>
      Don't Take The Money
    </glamorous(div)>
    <glamorous(div)>
      <glamorous(div)>
        Bleachers
      </glamorous(div)>
      <glamorous(div)>
        Gone Now
      </glamorous(div)>
    </glamorous(div)>
  </glamorous(div)>
</div>
`;
